taxcalc/tests/test_pufcsv.py

```diff
@@ -80,19 +80,21 @@ def test_agg(tests_path, puf_path):  # pylint: disable=redefined-outer-name
         raise ValueError(msg)
     # create aggregate diagnostic table using sub sample of records
     fullsample = pd.read_csv(puf_path)
-    rn_seed = 80  # to ensure two-percent sub-sample is always the same
-    subsample = fullsample.sample(frac=0.02,  # pylint: disable=no-member
+    rn_seed = 80  # to ensure sub-sample is always the same
+    subfrac = 0.02  # sub-sample fraction
+    subsample = fullsample.sample(frac=subfrac,  # pylint: disable=no-member
                                   random_state=rn_seed)
     rec_subsample = Records(data=subsample)
     calc_subsample = Calculator(policy=Policy(), records=rec_subsample)
     adt_subsample = multiyear_diagnostic_table(calc_subsample, num_years=nyrs)
     # compare combined tax liability from full and sub samples for each year
     taxes_subsample = adt_subsample.loc["Combined Liability ($b)"]
-    reltol = 0.01  # maximum allowed relative difference in tax liability
+    reltol = 0.04  # maximum allowed relative difference in tax liability
```

Comment:
@MattHJensen I ran the test function from my previous comment on the previous data set and taxcalc 0.8.5.  I got similar results to the ones produced with the newest data set and taxcalc 0.9.0:
>With different seeds, the same 2% sample rate, the same 0.04 tolerance and no weights, this test fails about 20% of the time. If I use the weights, this test fails every time with a relative difference ranging from about 0.07 to 0.15.

It seems like the seed that was chosen in the first test function yielded a sample that produced results very similar to those produced from the full data set.  I'm still not sure why using a weighted sample results in a larger relative difference than an unweighted sample.